File: PolicyPlus/EditPolKey.vb
Public Class EditPolKey
    Private Sub EditPolKey_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown
        If e.KeyCode = Keys.Escape Then DialogResult = DialogResult.Cancel
    End Sub
    Private Sub EditPolKey_Shown(sender As Object, e As EventArgs) Handles Me.Shown
        TextName.Select()
        TextName.SelectAll()
    End Sub
    Public Function PresentDialog(InitialName As String) As String
        TextName.Text = InitialName
        If ShowDialog() = DialogResult.OK Then
            Return TextName.Text
        Else
            Return ""
        End If
    End Function
End Class

File: PolicyPlus/EditPolKey.cs
using System;
using System.Windows.Forms;

namespace PolicyPlus
{
    public partial class EditPolKey
    {
        public EditPolKey()
        {
            InitializeComponent();
        }
        private void EditPolKey_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
                DialogResult = DialogResult.Cancel;
        }
        private void EditPolKey_Shown(object sender, EventArgs e)
        {
            TextName.Select();
            TextName.SelectAll();
        }
        public string PresentDialog(string InitialName)
        {
            TextName.Text = InitialName;
            if (ShowDialog() == DialogResult.OK)
            {
                return TextName.Text;
            }
            else
            {
                return "";
            }
        }
    }
}